apps/web/src/app/challenge/increment.action.ts

```diff
@@ -0,0 +1,43 @@
+'use server';
+import { getServerAuthSession } from '@repo/auth/server';
+import { prisma } from '@repo/db';
+import type { VoteType } from '@repo/db/types';
+
+export async function incrementOrDecrementUpvote(
+  id: number,
+  rootType: VoteType,
+  shouldIncrement: boolean,
```

Comment:
i know i did the original version but do we need this boolean. the existence of the record should imply to delete it right? cant remember if i had this here cause the user can spam clicks

Reply:
o wait u wrote this lolll I thought @jhechtf did. yeah I think that was what ur intention was, which is why I didn't touch it.

Reply:
ok maybe you wrote it 🤣

but i created the initial version at least

Reply:
does it still work without that boolean

Reply:
I think it should because we're checking if it exists or not